pkg/cli/importimage/importimage_test.go

```diff
@@ -573,6 +574,44 @@ func TestCreateImageImport(t *testing.T) {
 				ImportPolicy: imagev1.TagImportPolicy{Scheduled: true},
 			},
 		},
+		"import with importMode PreserveOriginal": {
+			name:       "testis",
+			importMode: imagev1.ImportModePreserveOriginal,
+			stream: &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{Name: "testis", Namespace: "other"},
+				Spec: imagev1.ImageStreamSpec{
+					Tags: []imagev1.TagReference{
+						{Name: "latest", From: &corev1.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:latest"}},
+					},
+				},
+			},
+			expectedImages: []imagev1.ImageImportSpec{{
+				From: corev1.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:latest"},
+				To:   &corev1.LocalObjectReference{Name: "latest"},
+				ImportPolicy: imagev1.TagImportPolicy{
+					ImportMode: imagev1.ImportModePreserveOriginal,
+				},
+			}},
+		},
+		"import with importMode Legacy": {
+			name:       "testis",
+			importMode: imagev1.ImportModeLegacy,
+			stream: &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{Name: "testis", Namespace: "other"},
+				Spec: imagev1.ImageStreamSpec{
+					Tags: []imagev1.TagReference{
+						{Name: "latest", From: &corev1.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:latest"}},
+					},
+				},
+			},
+			expectedImages: []imagev1.ImageImportSpec{{
+				From: corev1.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:latest"},
+				To:   &corev1.LocalObjectReference{Name: "latest"},
+				ImportPolicy: imagev1.TagImportPolicy{
+					ImportMode: imagev1.ImportModeLegacy,
+				},
+			}},
+		},
```

Comment:
Would you mind adding a test case also for `valid ImportMode values are %s or %s` error?

Reply:
I've added the test case in a new commit. I realized the test cases weren't running the validation step, so I've also added that step along with related error checking.